### PR TITLE
[Compute] `az vmss create`: Support set upgrade policy mode to automatic during flexible VMSS creation

### PR DESCRIPTION
**Pull request by Jing-song**
**Related command**

`az vmss create`

**Description**

support set upgrade policy to automatic during VMSS creation
close https://github.com/Azure/azure-cli/issues/29191

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py [hunk 1]
```diff
@@ -1309,18 +1307,17 @@ def build_vmss_resource(cmd, name, computer_name_prefix, location, tags, overpro
...
                                                          operation_group='virtual_machine_scale_sets'):
-        vmss_properties['upgradePolicy']['automaticOSUpgradePolicy'] = {}
-        automatic_os_upgrade_policy = vmss_properties['upgradePolicy']['automaticOSUpgradePolicy']
-
+        automatic_os_upgrade_policy = {}
         if enable_auto_os_upgrade is not None:
             automatic_os_upgrade_policy['enableAutomaticOSUpgrade'] = enable_auto_os_upgrade
 
-        if not automatic_os_upgrade_policy:
-            del automatic_os_upgrade_policy
```
Comment (Jing-song) [inline src/azure-cli/azure/cli/command_modules/vm/_template_builder.py:1323]: `del automatic_os_upgrade_policy` just deleted the variables, but key(`vmss_properties['upgradePolicy']['automaticOSUpgradePolicy']`) still exists

**Review by zhoxing-ms — APPROVED** [review]

**Comment by mimckitt** [comment]
Thanks for getting this fixed! When should I expect this functionality to be publicly available?

**Comment by yonzhan** [comment]
Build to Cloud Shell: 07/04/2024
Official Release: 07/09/2024

**Comment by mimckitt** [comment]
@yonzhan will it take some time to roll out broadly? I just tried to set the upgrade policy to automatic but it still ended up just creating a scale set with manual upgrade policy.

**Pull request merged**